priv/repo/migrations/20240715221724_create_get_hyperlinks_data_from_content.exs

```diff
@@ -0,0 +1,38 @@
+defmodule Oli.Repo.Migrations.CreateGetHyperlinksDataFromContent do
+  use Ecto.Migration
+
+  def up do
+    execute """
+    CREATE OR REPLACE FUNCTION get_hyperlink_references_from_revision(content JSONB)
+    RETURNS JSONB AS $$
+    DECLARE
+        result JSONB;
+    BEGIN
+        WITH RECURSIVE cte AS (
```

Comment:
I'm not sure that we need a recursive CTE in order to be able to query over a set of revision contents to find all hyperlinks.  Take a look at `lib/oli/qa/utils.ex`:

```
def elements_of_type(types, review) do
    project = Course.get_project!(review.project_id)
    publication_id = Publishing.project_working_publication(project.slug).id
    page_id = ResourceType.id_for_page()
    activity_id = ResourceType.id_for_activity()

    item_types =
      types
      |> Enum.map(&~s|@.type == "#{&1}"|)
      |> Enum.join(" || ")

    sql = """
    select
      rev.id,
      rev.title,
      jsonb_path_query(content, '$.** ? (#{item_types})')
    from published_resources as mapping
    join revisions as rev
    on mapping.revision_id = rev.id
    where mapping.publication_id = #{publication_id}
      and (rev.resource_type_id = #{page_id}
        or rev.resource_type_id = #{activity_id})
      and rev.deleted is false
    """
```

Reply:
As per your suggestion, I changed the CTE to use JSONPath. Additionally, I made some refactors to improve readability. Could you please take another look at your convenience?